File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        for(string str: strs){
            string newstr = str;
            sort(newstr.begin(), newstr.end());
            string key = newstr;
            mp[key].push_back(str);
        }

        vector<vector<string>> ans;
        for(auto it : mp){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = []
        mp = defaultdict(list)
        for str in strs:
            # key = "".join(sorted(str))
            # mp[key].append(str)
            count = [0] * 26
            for ch in str:
                count[ord(ch) - ord('a')] += 1
            
            mp[tuple(count)].append(str)
        
        # for val in mp.values():
        #     ans.append(val)
        
        # return ans

        return list(mp.values())
        
        